manifest: optional: remove sof from optional manifest

nashif opened the pull request:
Nothing in Zephyr uses SOF, it is the other way round, SOF uses
Zephyr, creating a cyclic dependency in some cases making it difficult
to apply changes to areas used by SOF upstream.

Part of #91061

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
manifest: optional: remove sof from optional manifest

Nothing in Zephyr uses SOF, it is the other way round, SOF uses
Zephyr, creating a cyclic dependency in some cases making it difficult
to apply changes to areas used by SOF upstream.

Part of #91061

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

lgirdwood reviewed (dismissed):
Ack - SOF depends on Zephyr not a cyclic dependency.

kv2019i commented:
@nashif wrote:
> @lyakh @kv2019i is this workaround still needed:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/d7c1ec670f2eb760c9670bc3d228553fb45a3c8b/soc/intel/intel_adsp/Kconfig.defconfig#L11

I had no idea we had such a workaround. Any idea who added this comment? Git-blame gives me just the gigantic HWMv2 commit.

It seems this is still depended on, builds will fail if I remove this.

LaurentiuM1234 commented:
@nashif do you want to have this for v4.3.0? If not, I'll schedule https://github.com/zephyrproject-rtos/zephyr/pull/97988 for v4.4.0

nashif committed:
intel_adsp: remove workaround for SOF setting core count

During transition to HWMv2 this workaround was added, which should
instead be in SOF and not in Zephyr, as CORE_COUNT is a SOF Kconfig.

Remove this and instead set the CORE_COUNT in SOF to the
MP_MAX_NUM_CPUS.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
west_commands: do not depend on CONFIG_SOF

Do not depend on SOF config, use RIMAGE_SCHEMA instead, defined in SOF.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
